llvm/lib/Target/SPIRV/SPIRVLegalizeImplicitBinding.cpp

```diff
@@ -112,11 +112,22 @@ uint32_t SPIRVLegalizeImplicitBinding::getAndReserveFirstUnusedBinding(
 }
 
 void SPIRVLegalizeImplicitBinding::replaceImplicitBindingCalls(Module &M) {
+  std::unordered_map<uint32_t, uint32_t> OrderIdToBinding;
```

Comment:
I don't think we need the hash table. The ImplicitBindingCalls should be sorted by the order id. You should have to only keep track of the last binding that you used.

Also, `std::unordered_map` is generally not used. In this case, where the orderIDs are dense, https://llvm.org/doxygen/classllvm_1_1DenseMap.html is probably the best option.